Add Figma to Vaadin feature description (#5192)
vaadin/docs: articles/tools/copilot/figma-to-code.adoc

---
title: Figma to Code
description: Generate Vaadin UI code from Figma designs using Vaadin Copilot.
order: 2
---

= [since:com.vaadin:vaadin@V24.5]#Figma to Code#

Figma to Vaadin is a feature of Vaadin Copilot that allows developers to copy designs from Figma and paste them directly into a Vaadin project, where Copilot automatically translates the design structure into Vaadin UI components.

== How does the Figma import work?

> The import process does not rely on AI interpretation of visual layouts. Copilot reads structured XML data from Figma’s clipboard and parses it into Vaadin UI code. This ensures that the same design always produces the same output.

Author: ```suggestion
The import process does not rely on AI interpretation of visual layouts. Copilot parses pasted Figma data into Vaadin UI code. This ensures that the same design always produces the same output.
```

The translation is deterministic and based on component metadata and layout structure defined in Figma.

== How do I use it?

. In Figma, select a frame or a section of a design.
. Copy it to the clipboard using kbd:[Ctrl+C] (Windows/Linux) or kbd:[Cmd+C] (macOS).
. Make sure your Vaadin project is running and Vaadin Copilot is enabled.
. In your project, navigate to the location where you want to insert the UI.

> . Paste using kbd:[Ctrl+V] or kbd:[Cmd+V].

Author: ```suggestion
. Select one of existing layout components and paste Figma data using kbd:[Ctrl+V] or kbd:[Cmd+V].
```

After a few seconds, the application refreshes and shows the generated UI.

You can copy an entire artboard or only a smaller section of a larger design.

== What are the limitations?

Out of the box, Figma to Vaadin generates functional UI components when designs use components from the *Vaadin Design System* Figma library.

> The translator relies on component metadata. For other component libraries, functional components are not automatically recognized unless translation rules are defined for them.

Reviewer: Not a real term. It should be something like "If translation rules are not defined, unrecognized Figma components will result static divs and spans instead of a web component.
